HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h

```diff
@@ -48,4 +48,47 @@ inline bool cudaCheck_(const char* file, int line, const char* cmd, cudaError_t
 
 #define cudaCheck(ARG) (cudaCheck_(__FILE__, __LINE__, #ARG, (ARG)))
 
+
+namespace {
+
+  template <typename T>
```

Comment:
@makortel, do you have any suggestions to reduce the copy paste from `cudaCheck()` ?

Reply:
I actually do have some ideas, let me play around a bit first.

Reply:
@fwyzard I came up with this
https://github.com/makortel/cmssw/blob/patatrack_pr414/HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h

Further possibility would be to pass the `CUresult`/`cudaError_t` all the way up to `formatCudaError`, but I wanted to share this "earlier" version and leave further tuning to you discretion.